src/percy-agent-client/dom.ts

```diff
@@ -225,6 +226,38 @@ class DOM {
     })
   }
 
+  /**
+   * Capture in-memory canvas elements & serialize them to images into the
+   * cloned DOM.
+   *
+   * Without this, applications that have canvas elements will be missing and
+   * appear broken. The Canvas DOM API allows you to covert them to images, which
+   * is what we're doing here to capture that in-memory state & serialize it
+   * into the DOM Percy captures.
+   *
+   * It's important to note the `.toDataURL` API requires WebGL canvas elements
+   * to use `preserveDrawingBuffer: true`. This is because `.toDataURL` captures
+   * from the drawing buffer, which is cleared after each render by default for
+   * performance.
+   *
+   */
+  private serializeCanvasElements(clonedDOM: HTMLDocument): void {
+    function canvasToImage($canvas: any) {
```

Comment:
We probably don't want to do this if JS is enabled, or do we?

Reply:
OH, excellent question!! I'd say probably not 🤔 The user would need to make sure it can render properly? That's can of worms.

Hm.